test/functional/insert/insert_spec.lua

```diff
@@ -0,0 +1,41 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear, feed, insert = helpers.clear, helpers.feed, helpers.insert
+local command = helpers.command
+local eq = helpers.eq
+local expect = helpers.expect
+local funcs = helpers.funcs
+
+describe('insert-mode', function()
+  before_each(function()
+    clear()
+  end)
+
+  it('CTRL-@', function()
+    -- Inserts last-inserted text, leaves insert-mode.
+    insert('hello')
+    feed('i<C-@>x')
+    expect('hellhello')
+
+    -- C-Space is the same as C-@.
+    -- CTRL-SPC inserts last-inserted text, leaves insert-mode.
+    feed('i<C-Space>x')
+    expect('hellhellhello')
+
+    -- CTRL-A inserts last inserted text
+    feed('i<C-A>x')
+    expect('hellhellhellhelloxo')
+  end)
+
+  it('ALT/META #8213', function()
+    -- Mapped ALT-chord behaves as mapped.
+    command('inoremap <M-l> meta-l')
+    command('inoremap <A-j> alt-j')
+    feed('i<M-l> xxx <A-j><M-h><A-h>')
```

Comment:
Shouldn't it be `<M-h>a<A-h>` here?  The intent is only to treat Meta/Alt chords like Esc when in insert mode, so `<M-h>` would go to normal mode and move left.  Then `<A-h>` would happen in normal mode and do nothing, since it isn't mapped.

Reply:
Well, your commit message says it's for insert mode as do the doc changes.  :) However, consistent behavior among modes is easier to wrap your head around.  So do the docs change or the implementation?

Reply:
My previous comment was confusing.  The TODO is "totally ignore unmapped ALT chords in normal mode".

Not sure what's consistent.  With the current PR, unmapped ALT chords will (accidentally) behave like ESC+key, in normal-mode.

Reply:
> Not sure what's consistent. With the current PR, unmapped ALT chords will (accidentally) behave like ESC+key, in normal-mode.

Right, which is consistent with the behavior in insert mode.  Is there a reason to _restrict_ this to insert mode?  Doesn't that just add confusion?